python/setup.py

```diff
@@ -63,22 +63,32 @@ def get_cuda_version_from_header(cuda_include_dir):
 # valid symbols for specific version of CUDA.
 
 cwd = os.getcwd()
-preprocess_files = ["gpu.pxd"]
-supported_cuda_versions = {"10.1", "10.2", "11.0"}
-
-for file_p in preprocess_files:
-    pxi_file = ".".join(file_p.split(".")[:-1])
-    pxi_file = pxi_file + ".pxi"
-
-    if CUDA_VERSION in supported_cuda_versions:
-        shutil.copyfile(
-            os.path.join(cwd, "rmm/_cuda", CUDA_VERSION, pxi_file),
-            os.path.join(cwd, "rmm/_cuda", file_p),
+files_to_preprocess = ["gpu.pxd"]
+
+# The .pxi file is unchanged between some CUDA versions
+# (e.g., 11.0 & 11.1), so we keep only a single copy
+# of it
+cuda_version_to_pxi_dir = {
+    "10.1": "10.1",
+    "10.2": "10.2",
+    "11.0": "11.0",
+    "11.1": "11.0",
+    "11.2": "11.0",
+}
+for pxd_basename in files_to_preprocess:
+    pxi_basename = os.path.splitext(pxd_basename)[0] + ".pxi"
+    if CUDA_VERSION in cuda_version_to_pxi_dir:
+        pxi_pathname = os.path.join(
+            cwd,
+            "rmm/_cuda",
+            cuda_version_to_pxi_dir[CUDA_VERSION],
+            pxi_basename,
         )
+        pxd_pathname = os.path.join(cwd, "rmm/_cuda", pxd_basename)
+        shutil.copyfile(pxi_pathname, pxd_pathname)
```

Comment:
Out of scope of this PR, but... since we always copy the file it always invalidates the cache and forces rebuild. Could we update this to read the file and check for the version and then only copy if it needs to be updated?

Reply:
Good catch -- we need to do more than just check the version though. If we later change the `.pxi` file and the `.pxd` already exists, then we can't force a rebuild without manually rm'ing the `.pxd`. Thus, I added logic to compare the `.pxi` and `.pxd` files before deciding if we need a copy.